File: JavaScript/DailyCodeChallenge/251104_imageSearch.js
function imageSearch(images, term) {
  const regex = new RegExp(`${term}`,`ig`)


  const returnArr = images.filter(image => image.match(regex))

  console.log(returnArr)
  return returnArr;
}


imageSearch(["cat.jpg", "dogToy.jpeg", "kitty-cat.png", "catNip.jpeg", "franken_cat.gif"], "Cat")


/*
Image Search
On November 4th, 2001, Google launched its image search, allowing people to find images using search terms. In this challenge, you will imitate the image search.

Given an array of image names and a search term, return an array of image names containing the search term.

Ignore the case when matching the search terms.
Return the images in the same order they appear in the input array.

*/

File: Python/Daily_Code_Challenge/251104-image-search.py
def image_search(images, term):

    found = [image for image in images if term.lower() in image.lower()]

    print(found)

    return found


image_search(["cat.jpg", "dogToy.jpeg", "kitty-cat.png", "catNip.jpeg", "franken_cat.gif"], "Cat")

'''
Image Search

On November 4th, 2001, Google launched its image search, allowing people to find images using search terms. In this challenge, you will imitate the image search.

Given an array of image names and a search term, return an array of image names containing the search term.

    Ignore the case when matching the search terms.
    Return the images in the same order they appear in the input array.

Tests

    Passed: 1. image_search(["dog.png", "cat.jpg", "parrot.jpeg"], "dog") should return ["dog.png"].
    Passed: 2. image_search(["Sunset.jpg", "Beach.png", "sunflower.jpeg"], "sun") should return ["Sunset.jpg", "sunflower.jpeg"].
    Passed: 3. image_search(["Moon.png", "sun.jpeg", "stars.png"], "PNG") should return ["Moon.png", "stars.png"].
    Passed: 4. image_search(["cat.jpg", "dogToy.jpeg", "kitty-cat.png", "catNip.jpeg", "franken_cat.gif"], "Cat") should return ["cat.jpg", "kitty-cat.png", "catNip.jpeg", "franken_cat.gif"].
'''